jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -500,52 +513,61 @@ public boolean getReportDirs()
      *
      * @param listener the listener to add
      */
-    public synchronized void addListener(Listener listener)
+    public void addListener(Listener listener)
     {
         if (listener == null)
             return;
-        _listeners.add(listener);
+        try (AutoLock l = _lock.lock())
+        {
+            _listeners.add(listener);
+        }
     }
 
     /**
      * Remove a registered listener
      *
      * @param listener the Listener to be removed
      */
-    public synchronized void removeListener(Listener listener)
+    public void removeListener(Listener listener)
     {
         if (listener == null)
             return;
-        _listeners.remove(listener);
+        try (AutoLock l = _lock.lock())
+        {
+            _listeners.remove(listener);
+        }
     }
 
     /**
      * Start the scanning action.
      */
     @Override
-    public synchronized void doStart()
+    public void doStart()
     {
-        if (_running)
-            return;
+        try (AutoLock l = _lock.lock())
```

Comment:
I'm not really sure we need this lock.... but best to keep logic as is for this PR.